providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -833,3 +1024,32 @@ const OSSL_DISPATCH ec_keymgmt_functions[] = {
       (void (*)(void))ec_query_operation_name },
     { 0, NULL }
 };
+
+#ifndef OPENSSL_NO_SM2
+const OSSL_DISPATCH sm2_keymgmt_functions[] = {
+    { OSSL_FUNC_KEYMGMT_NEW, (void (*)(void))ec_newdata },
+    { OSSL_FUNC_KEYMGMT_GEN_INIT, (void (*)(void))ec_gen_init },
+    { OSSL_FUNC_KEYMGMT_GEN_SET_TEMPLATE,
+      (void (*)(void))ec_gen_set_template },
+    { OSSL_FUNC_KEYMGMT_GEN_SET_PARAMS, (void (*)(void))ec_gen_set_params },
+    { OSSL_FUNC_KEYMGMT_GEN_SETTABLE_PARAMS,
+      (void (*)(void))ec_gen_settable_params },
+    { OSSL_FUNC_KEYMGMT_GEN, (void (*)(void))sm2_gen },
+    { OSSL_FUNC_KEYMGMT_GEN_CLEANUP, (void (*)(void))ec_gen_cleanup },
+    { OSSL_FUNC_KEYMGMT_FREE, (void (*)(void))ec_freedata },
+    { OSSL_FUNC_KEYMGMT_GET_PARAMS, (void (*) (void))sm2_get_params },
+    { OSSL_FUNC_KEYMGMT_GETTABLE_PARAMS, (void (*) (void))sm2_gettable_params },
+    { OSSL_FUNC_KEYMGMT_SET_PARAMS, (void (*) (void))ec_set_params },
```

Comment:
Strictly speaking `ec_set_params` also supports `OSSL_PKEY_PARAM_USE_COFACTOR_ECDH` which is not declared by `sm2_settable_params`. That doesn't seem quite right.

Reply:
What does this `OSSL_PKEY_PARAM_USE_COFACTOR_ECDH` flag use for?

Reply:
That's specific to ECDH, so not in scope for SM2.  If you're still curious, I'm sure someone can explain it, but that should happen *off* this PR.

Reply:
> Strictly speaking ec_set_params also supports OSSL_PKEY_PARAM_USE_COFACTOR_ECDH which is not declared by sm2_settable_params. That doesn't seem quite right.

This seems no harmful for the actual functionality, right?

Reply:
No it's not